routers/repo/pull.go

```diff
@@ -69,11 +71,23 @@ func getForkRepository(ctx *context.Context) *models.Repository {
 	ctx.Data["ForkFrom"] = forkRepo.Owner.Name + "/" + forkRepo.Name
 	ctx.Data["ForkFromOwnerID"] = forkRepo.Owner.ID
 
-	if err := ctx.User.GetOrganizations(true); err != nil {
-		ctx.Handle(500, "GetOrganizations", err)
+	if err := ctx.User.GetOwnedOrganizations(); err != nil {
+		ctx.Handle(500, "GetOwnedOrganizations", err)
 		return nil
 	}
-	ctx.Data["Orgs"] = ctx.User.Orgs
+	var orgs []*models.User
+	for _, org := range ctx.User.OwnedOrgs {
+		if forkRepo.OwnerID != org.ID && !org.HasForkedRepo(forkRepo.ID) {
+			orgs = append(orgs, org)
+		}
+	}
+	ctx.Data["Orgs"] = orgs
+
+	if canForkToUser {
+		ctx.Data["ContextUser"] = ctx.User
```

Comment:
In both places where `getForkRepository()` is called (`Fork` and `ForkPost` in routers/repo/pull.go), `ctx.Data["ContextUser"]` is immediately overwritten after this function is called.

Reply:
@ethantkoenig it is not overwritten in `Fork` handler only in `ForkPost`. It is intentional so - `getForkRepository` fills in default `ctx.Data["ContextUser"]` value and in `ForkPost` it is overwritten by actually selected one from dropdown box.

Reply:
Oh, I missed that you had removed that line